src/pudl/transform/ferc1.py

```diff
@@ -793,17 +808,33 @@ def check_table_calculations(
     ]
     calced_values = calculated_df[(calculated_df.abs_diff.notnull())]
     off_ratio = len(off_df) / len(calced_values)
-    if off_ratio > 0:
-        logger.warning(
-            f"{table_name}: has {len(off_df)} ({off_ratio:.02%}) records that don't "
-            "calculate exactly."
-        )
+
     if off_ratio > params.calculation_tolerance:
         raise AssertionError(
             f"Calculations in {table_name} are off by {off_ratio}. Expected tolerance "
             f"of {params.calculation_tolerance}."
         )
 
+    # We'll only get here if the proportion of calculations that are off is acceptable
+    if off_ratio > 0:
+        logger.info(
+            f"{table_name}: has {len(off_df)} ({off_ratio:.02%}) records whose "
+            "calculations don't match. Adding correction records to make calculations "
+            "match reported values."
+        )
+        corrections = off_df.copy()
+        corrections[params.column_to_check] = (
+            corrections[params.column_to_check].fillna(0.0)
+            - corrections["calculated_amount"]
+        )
+        corrections[xbrl_factoid_name] = corrections[xbrl_factoid_name] + "_correction"
+        corrections["row_type_xbrl"] = "correction"
+        corrections["record_id"] = pd.NA
+
+        calculated_df = pd.concat(
+            [calculated_df, corrections], axis="index"
+        ).reset_index()
```

Comment:
are you sure the index will match perfectly here? this seems more risky than merging on the `pks_wo_factoid` with some merge validation

Reply:
hm nvm!